eeps/eep-0064.md

```diff
@@ -0,0 +1,395 @@
+
+A later step, preferably done by the parser, uses the white-space
+on the ending line as the definition of the string's indentation
+and strips that particular white-space sequence from every line
+in the string, and strips the newline preceding the ending line.
+
+If any of the lines do not start with the defined indentation
+either because the line is too short or if the prefix differs,
+the parser reports a syntax error.
+
+Requiring that all lines must have exactly the same indentation
+characters is a simple solution to not have to define how indentation
+white-space (tab vs. space) normalization should be done,
+and also seems like a reasonable requirement.
+
+### `CR`, `LF` and White-space
+
+The characters `CR` - Unicode code-point value 13,
+`LF` - code-point 10, and white-space - as as defined
+in the Erlang scanner today, are handled as usual by the scanner,
+except that if the line preceding the ending line
+ends in `CR` `LF` then also the `CR` is interpreted
+as part of the newline and is stripped along with the `LF`.
+This is a convenience for systems with `CR` `LF` newlines.
+
+Other than that, `CR`, `LF` and white-space within the string
+is passed through as-is.
+
+This means that the examples in the following text
+where a normal string is used as matching reference
+assumes that the source code has `LF`-only newlines,
+such as:
+
+    "\nX" = """
+        
+        X
+        """"
+
+If the source code has `CR` `LF` newlines that
+example instead becomes:
+
+    "\r\nX" = """
+        
+        X
+        """"
+
+### Leading and trailing newline
+
+The rules above strips one leading and one trailing newline.
+This is a simple convention that also gives control over
+the string's content:
+
+    "\n  X\n" = """
+        
+          X
+        
+        """,
+    
+    "X" = """
+        X
+        """,
+    
+    "" = """
+         
+         """
+
+Note that the following could be a syntax error; too short
+multi-line string since trailing newline shall be stripped
+both from the starting line and from the last content line,
+so the content could be seen as less than empty.
+But it is more convenient to allow this as also an empty string:
+
+    "" = """
+         """
+
+### Indentation
+
+The rules above facilitates indentation of the content
+to adhere to the surrounding code.  The ending line
+determines the indentation.
+
+    """
+    This text
+    has no indentation
+    """ =
+        "This text\nhas no indentation",
+    
+    """
+        This text
+        has indentation
+    """ =
+        "    This text\n    has indentation",
+    
+    """
+      This text
+    has an indented first line
+    """ =
+        "  This text\nhas an indented first line",
+    
+    """
+    This is a syntax error (incorrect indentation)
+        """,
+    
+    """ This is a syntax error
+    (non-white-space on start line)
+    """,
+    
+    """
+    This will probably be a syntax error
+    since the ending line is not here"""
```

Comment:
```
    """
    This will probably be a syntax error
    since the ending line is not here"""
```
IMO this is not totally correct. It's a syntax error, but the shell would wait for the string ending, and in a module the parser would give up, for example:
```erlang
-module(foo).
-export([foo/0]).

foo() ->
    """
    This will cause the parser to give up
    since the ending line is not here""".
```

```suggestion
    """
    This will cause the shell to wait for the `"""` in a new line or 
    the parser to give up in a module, since the ending line is not here"""
```

Reply:
I will rephrase that